### PR TITLE
[feature] add support for managing tfe providers

### PR DESCRIPTION
**Pull request by ryanking**
### Test Plan
* CI
* manually tested on 1 internal repo

### References
* https://www.terraform.io/docs/providers/tfe/index.html

### CODE REVIEW

File: plan/plan_test.go [hunk 1]
```diff
@@ -199,3 +199,24 @@ func Test_buildTerraformIgnore(t *testing.T) {
...
+
+	plan := buildPlan(t, "tfe_provider_yaml")
+
+	on := func(c ComponentCommon) {
+		r.NotNil(c)
+		r.NotNil(c.Providers.Tfe)
+		r.True(c.Providers.Tfe.Enabled)
+	}
+
+	off := func(c ComponentCommon, enabled bool) {
```
Comment (edulop91) [inline plan/plan_test.go:214]: enabled unused

File: templates/account/fogg.tf.tmpl [hunk 2]
```diff
@@ -47,6 +47,11 @@
   {{ template "datadog_provider" .Providers.Datadog }}
 {{ end }}
 
+{{ if .Providers.Tfe }}{{ if .Providers.Tfe.Enabled }}
```
Comment (edulop91) [inline templates/account/fogg.tf.tmpl:50]: wonder if these conditionals should be inside the tfe template so theyre not repeated every fogg.tf template

File: plan/plan_test.go (hunk 1 above)
Comment (ryanking) [reply plan/plan_test.go:214]: went with enabled/disabled

File: templates/account/fogg.tf.tmpl (hunk 2 above)
Comment (ryanking) [reply templates/account/fogg.tf.tmpl:50]: That makes sense, but given that the rest of the them follow this pattern, I would like to leave it as is for the sake of consistency and address it in a follow-up #462

**Pull request merged**